### PR TITLE
Feature/steps crud

### PR DESCRIPTION
**Pull request by miatrinity**
Hey please check this out

### CODE REVIEW

File: app/views/steps/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<section class="section">
+  <div class="columns">
+    <div class="column">
+      <h1 class="title ">Title of the Routine - Steps</h1>
```
Comment (MiaSinek) [inline app/views/steps/index.html.erb]: Fix this

File: app/views/steps/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<%= form_with model: [routine, step], local: true do |form| %>
+  <% if step.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(step.errors.count, "error") %> prohibited this step from being saved:</h2>
```
Comment (MiaSinek) [inline app/views/steps/_form.html.erb:4]: Add test: error msg if not saved

File: app/views/steps/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<section class="section">
+  <div class="columns">
+    <div class="column">
+      <nav class="breadcrumb">
+        <ul>
+          <li>
+            <a href="#">Routine X</a>
```
Comment (MiaSinek) [inline app/views/steps/new.html.erb]: Fix this

**Review by MiaSinek — CHANGES_REQUESTED** [review]
- Test error msg if step can't be saved or edited
- Add dependent :destroy to routine.rb
- consider giving _top_stuff.html.erb a meaningful name :)

Other comments: see code

File: app/views/steps/index.html.erb (hunk 1 above)
Comment (miatrinity) [reply app/views/steps/index.html.erb]: Done!

File: app/views/steps/_form.html.erb (hunk 2 above)
Comment (miatrinity) [reply app/views/steps/_form.html.erb:4]: Won't test - HTML 5 generates error message that doesn't allow the form to be submitted

File: app/views/steps/new.html.erb (hunk 3 above)
Comment (miatrinity) [reply app/views/steps/new.html.erb]: Fixed!

**Pull request merged**